### ISSUE TITLE
Some types take away waay too much space in inspector.

### ISSUE THREAD

**Issue by scayze** [godotengine/godot#19830]
Godot 3.1dev

Certain properties take away too much space in the new inspector, making it a real pain to use.
Especially types like Vector3 where all 3 components are put vertically above eachother instead of beside eachother.

Even worse is the transform type. Not even once i had to change the transform manually in the inspector. Do we really need that matrix of 3x4 in the inspector for it? cluttering the view with unnecessary information?

In comparison to the old inspector, which was nice and compact, this seems like a total waste space:

![image](https://user-images.githubusercontent.com/10299514/42056284-1f399b28-7b1a-11e8-956f-be47c9a23029.png)
![image](https://user-images.githubusercontent.com/10299514/42056287-21ca6840-7b1a-11e8-81b2-85a3b31a5784.png)

the old one about three times as compact.

solutions could be to put the attributes horizontally again, or to be able to collapse seperate types (which id dislike since it would make usability worse imo).

It would already help to remove the transform type from the inspector altogether since i don't think anyone is actually using it, but i might be wrong here.

**Comment by toger5** [comment]
> It would already help to remove the transform type from the inspector altogether since i don't think anyone is actually using it, but i might be wrong here.

This actually could be a good idea. The situation I use it for, is to figure out how transforms work. by changing translation, rotation and checking how the transform values change to than implement something in gdscript.

> In comparison to the old inspector, which was nice and compact, this seems like a total waste space

I highly disagree. The editor has the functionality of setting props AND reading the current status.
And both of those core functionalities get highly improved. It is not possible to actually read all values (see oration in the old version) and it takes much less time to edit them.

the current size of the transform section is also the average size of the inspector dock.
Since you usually only work on transform, and than on another prop it is perfect if the transform takes all the dock area. scrolling is really easy to execute since you don't need to move your mouse to a precise location. so reaching another part of the Inspector is still really easy.

**Comment by scayze** [comment]
I agree that readability used to be pretty bad with the old inspector, considering it wasnt even able to read all values at once. 

I didn't say i want to revert to that design. i dont want that either! I just ment to say that attributes ordered vertically above each other create a huge unused space:
![image](https://user-images.githubusercontent.com/10299514/42126767-2cedd116-7c8d-11e8-8b11-d7ddda417ed2.png)

Ordering them horizontal, beneath the name of the element would use that space way more efficiant
![image](https://user-images.githubusercontent.com/10299514/42126775-437d616c-7c8d-11e8-8682-659ff2e30c9d.png)

it doesn't decrease the clickable area or font size or anything, so it would hurt usability/readability at all, while still saving lots of space making the inspector more compact.


> the current size of the transform section is also the average size of the inspector dock.
Since you usually only work on transform, and than on another prop it is perfect if the transform takes all the dock area. scrolling is really easy to execute since you don't need to move your mouse to a precise location. so reaching another part of the Inspector is still really easy.

I strongly disagree that the user only works on the transform when editing an object(or generally speaking, only working on a single category).
more than often i have lots of attributes to edit, al in different categories, especially when working with materials or sub-ressources in general.

Since the new inspector is here, i find myself collapsing/expanding categories and furiously scrolling up and down trying to find the attributes i need.

The editor being compact should be a top priority imo.

@toger5

**Comment by smt923** [comment]
I agree, the new transform it's self, I think is a perfectly acceptable balance of showing all the values being editable right there even though it takes more space in the inspector, but the other values should be stacked similarly too imo - I recently wrote a comment about this in another issue: https://github.com/godotengine/godot/issues/19084#issuecomment-400710026
